app/src/main/java/com/example/plantsapp/presentation/ui/notification/NotificationWorker.kt

```diff
@@ -0,0 +1,119 @@
+package com.example.plantsapp.presentation.ui.notification
+
+import android.app.NotificationChannel
+import android.app.NotificationManager
+import android.content.Context
+import android.os.Build
+import androidx.core.app.NotificationCompat
+import androidx.core.app.NotificationManagerCompat
+import androidx.hilt.work.HiltWorker
+import androidx.work.Worker
+import androidx.work.WorkerParameters
+import com.bumptech.glide.Glide
+import com.example.plantsapp.R
+import com.example.plantsapp.domain.model.Plant
+import com.example.plantsapp.domain.model.Task
+import com.example.plantsapp.domain.repository.TasksRepository
+import dagger.assisted.Assisted
+import dagger.assisted.AssistedInject
+import kotlinx.coroutines.flow.first
+import kotlinx.coroutines.runBlocking
+import java.util.Date
+
+@HiltWorker
+class NotificationWorker @AssistedInject constructor(
+    @Assisted private val context: Context,
+    @Assisted workerParams: WorkerParameters,
+    private val repository: TasksRepository
+) : Worker(context, workerParams) {
+
+    private var notificationId = 0
+
+    override fun doWork(): Result {
+        runBlocking {
+            val plantsWithTasks =
+                repository.getPlantsWithTasksForDate(Date()).first()
+
+            plantsWithTasks.forEach { (plant, tasks) ->
+                sendTaskNotifications(plant, tasks)
+            }
+        }
+
+        return Result.success()
+    }
+
+    private fun sendTaskNotifications(
+        plant: Plant,
+        tasks: List<Task>
+    ) {
+        createChannel(context)
+
+        val notificationPicture = Glide.with(context)
+            .asBitmap()
+            .load(plant.plantPicture ?: R.drawable.ic_baseline_image_24)
+            .submit()
+            .get()
+
+        val notifications = tasks.map {
+            NotificationCompat.Builder(context, NOTIFICATION_CHANNEL_ID)
+                .setSmallIcon(R.drawable.ic_plant_24)
+                .setContentTitle(plant.name.value)
+                .setContentText(
+                    context.getNotificationText(
+                        plantName = plant.name.value,
+                        task = it
+                    )
+                )
+                .setLargeIcon(notificationPicture)
+                .setGroup(NOTIFICATION_GROUP_KEY) // TODO make it unique for each plant
```

Comment:
How about using Plant.Name here as key for Group? (As long as Plant.Name us unique)